---
title: "Sampling, sheep"
date: 'Last updated: May 17 2023'
output:
  word_document: default
  pdf_document: default
---

This Markdown notebook describes the process of empirically sampling an amplicon to determine the margin of error associated with a particular number of CG calls (reads) per sample. Using a for loop, this script takes random samplings of a real amplicon sample in read sizes between 1 and n, calls the methylation of that sample based on the randomly sampled data, and plots it. From this, we are able to infer an suitable number of CG calls to aim for (i.e. threshold) while allowing an appropriate amount of error.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Load required packages
```{r}
library(dplyr)
library(ggplot2)
library(cowplot)
library(reshape2)
library(Manu)
```

# Sheep
## Low methylation
### 1 Ram_Ear_B5_EARNS_2013
### 2 Ram_Ear_D1_YELLOW
### 3 Ram_Ear_E5_EARNS_2013
### 4 Ram_Ear_99_Lammerlaw_2013

## Average methylation
### 5 Ram_Ear_E6_MATAK_2016
### 6 Ram_Ear_B12_EARNS_2016
### 7 Ram_Ear_D10_EARNS_2015

## High methylation
### 8 BHClock_3G10_2010
### 9 Wether_Ear_E3_2014
### 10 Wether_Ear_A2_EARNS_2013

# Data preparation
```{r, echo=FALSE}
data_low_1 <- read.delim("/Users/victoriasugrue/Dropbox (Otago University)/MKLN1_AmpSampling/CpG_context_Ram_Ear_B5_EARNS_2013_AllSequencingConcatenatedFeb2023_R1_trimmed.fq_bismark.txt", skip=1)
# Rename columns
colnames(data_low_1) <- c("Sample", "Strand", "Locus", "Location", "MethylationCall")
#data_low_1$MethylationCall[1:100] # prints on the first 100

data_low_2 <- read.delim("/Users/victoriasugrue/Dropbox (Otago University)/MKLN1_AmpSampling/CpG_context_Ram_Ear_D1_YELLOW_AllSequencingConcatenatedFeb2023_R1_trimmed.fq_bismark.txt", skip=1)
# Rename columns
colnames(data_low_2) <- c("Sample", "Strand", "Locus", "Location", "MethylationCall")
#data_low_2$MethylationCall[1:100] # prints on the first 100

data_low_3 <- read.delim("/Users/victoriasugrue/Dropbox (Otago University)/MKLN1_AmpSampling/CpG_context_Ram_Ear_E5_EARNS_2013_AllSequencingConcatenatedFeb2023_R1_trimmed.fq_bismark.txt", skip=1)
# Rename columns
colnames(data_low_3) <- c("Sample", "Strand", "Locus", "Location", "MethylationCall")
#data_low_3$MethylationCall[1:100] # prints on the first 100

data_low_4 <- read.delim("/Users/victoriasugrue/Dropbox (Otago University)/MKLN1_AmpSampling/CpG_context_Ram_Ear_99_LAMMERLAW_2015_AllSequencingConcatenatedFeb2023_R1_trimmed.fq_bismark.txt", skip=1)
# Rename columns
colnames(data_low_4) <- c("Sample", "Strand", "Locus", "Location", "MethylationCall")
#data_low_4$MethylationCall[1:100] # prints on the first 100

data_mid_5 <- read.delim("/Users/victoriasugrue/Dropbox (Otago University)/MKLN1_AmpSampling/CpG_context_Ram_Ear_E6_MATAK_2016_AllSequencingConcatenatedFeb2023_R1_trimmed.fq_bismark.txt", skip=1)
# Rename columns
colnames(data_mid_5) <- c("Sample", "Strand", "Locus", "Location", "MethylationCall")
#data_mid_5$MethylationCall[1:100] # prints on the first 100

data_mid_6 <- read.delim("/Users/victoriasugrue/Dropbox (Otago University)/MKLN1_AmpSampling/CpG_context_Ram_Ear_B12_EARNS_2016_AllSequencingConcatenatedFeb2023_R1_trimmed.fq_bismark.txt", skip=1)
# Rename columns
colnames(data_mid_6) <- c("Sample", "Strand", "Locus", "Location", "MethylationCall")
#data_mid_6$MethylationCall[1:100] # prints on the first 100

data_mid_7 <- read.delim("/Users/victoriasugrue/Dropbox (Otago University)/MKLN1_AmpSampling/CpG_context_Ram_Ear_D10_EARNS_2015_AllSequencingConcatenatedFeb2023_R1_trimmed.fq_bismark.txt", skip=1)
# Rename columns
colnames(data_mid_7) <- c("Sample", "Strand", "Locus", "Location", "MethylationCall")
#data_mid_7$MethylationCall[1:100] # prints on the first 100

data_high_8 <- read.delim("/Users/victoriasugrue/Dropbox (Otago University)/MKLN1_AmpSampling/CpG_context_BHClock1_3G10_2010_AllSequencingConcatenatedFeb2023_R1_trimmed.fq_bismark.txt", skip=1)
# Rename columns
colnames(data_high_8) <- c("Sample", "Strand", "Locus", "Location", "MethylationCall")
#data_high_8$MethylationCall[1:100] # prints on the first 100

data_high_9 <- read.delim("/Users/victoriasugrue/Dropbox (Otago University)/MKLN1_AmpSampling/CpG_context_Wether_Ear_E3_2014_AllSequencingConcatenatedFeb2023_R1_trimmed.fq_bismark.txt", skip=1)
# Rename columns
colnames(data_high_9) <- c("Sample", "Strand", "Locus", "Location", "MethylationCall")
#data_high_9$MethylationCall[1:100] # prints on the first 100

data_high_10 <- read.delim("/Users/victoriasugrue/Dropbox (Otago University)/MKLN1_AmpSampling/CpG_context_Wether_Ear_A2_EARNS_2013_AllSequencingConcatenatedFeb2023_R1_trimmed.fq_bismark.txt", skip=1)
# Rename columns
colnames(data_high_10) <- c("Sample", "Strand", "Locus", "Location", "MethylationCall")
#data_high_10$MethylationCall[1:100] # prints on the first 100
```

# Low 1 

# Create list for data to be dumped into
```{r}
sample_data_list_1 <- c()
```

# Sampling loop - creates random samples of sizes between 1 and n
```{r, message=F}
# Create an empty data frame to store the results
results_df_1 <- data.frame(Sample_Size = numeric(), Methylation = numeric())

for (i in c(10, 50, 100, 150, 200, 250, 300, 350, 400, 450, 500, 550, 600, 650, 700)) {
  for (j in 1:1000) {
    sample_data_1 <- sample_n(data_low_1, i)  
    methylation_table <- table(sample_data_1$MethylationCall)
    methylation <- as.numeric(methylation_table["Z"] / sum(methylation_table))
    sample_data_list_1 <- cbind(Sample_Size = i, Methylation = methylation)
    results_df_1 <- rbind(results_df_1, sample_data_list_1)
  }
}
```


# Plot
```{r}
methylation_plot_1 <- ggplot(results_df_1, aes(x=Sample_Size, y=Methylation)) +
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Methylation (%)") +
  xlab("CG calls assayed")
methylation_plot_1
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Methylation_1.png",methylation_plot_1, width=10, height = 5, dpi=1000)
```


# Finding most representative methylation call of this dataset
```{r}
mean_1 <- mean(results_df_1$Methylation[14001:15000])
```

# Making a list of the error to plot (percentage of 100%)

```{r}
results_df_1$error <- as.numeric(abs(mean_1-results_df_1$Methylation))

error_plot_1 <- ggplot(results_df_1, aes(x=Sample_Size, y=error)) + 
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Margin of Error") +
  xlab("CG calls assayed")
error_plot_1
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Error_1.png",error_plot_1, width=10, height = 5, dpi=1000)
```

# Find 99.5% confidence interval
```{r}
cis_1 <- aggregate(error ~ Sample_Size, data = results_df_1, function(x) quantile(x, c(0.0025, 0.9975)))
upper_cis_1 <- cis_1$error[,2]

df_1 <- data.frame(Sample_Size = cis_1$Sample_Size,
                 Upper_CI = upper_cis_1)
```


# Making a list of the error to plot (+/- pp)
```{r}
#establishing modelling parameters
zstar = qnorm(.995) #This defines 99.5% confidence level
p = mean_1 #The approximate methylation level from this example
n = c(10, 50, 100, 150, 200, 250, 300, 350, 400, 450, 500, 550, 600, 650, 700) #Numbers of cytosine calls assayed
E = zstar*sqrt((p*(1-p))/n) #Margin of error calculation given above variables

#coercing into dataframe
Edf_1 <- as.data.frame(E)
ndf_1 <- as.data.frame(n)
Edf_1$n <- ndf_1$n
Edf_1$Percent <- Edf_1$E*100

#establishing intercepts
line5 <- 5
line25 <- 2.5
line1 <- 1

model_plot_1 <- ggplot(Edf_1, aes(x=n, y=Percent)) +
  geom_point(alpha=2/5) +
  geom_hline(yintercept = line5, colour="#CC0000", linetype=2) +
  geom_text(aes(737,line5,label = "5%", vjust = -0.2, colour="#006600")) +
  geom_hline(yintercept = line25, colour="#006600", linetype=2) +
  geom_text(aes(737,line25,label = "2.5%", vjust = -0.2, colour="#0066CC")) +
  geom_hline(yintercept = line1, colour="#0066CC",  linetype=2) +
  geom_text(aes(737,line1,label = "1%", vjust = -0.2, colour="#CC0000")) +
  labs(x="Number of Cs in CG context", y="Margin of error") +
  theme_light() +
  ylim(0,100) +
  theme(legend.position="none")  
model_plot_1
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingModel_1.png",model_plot_1, width=10, height = 5, dpi=1000)

```


# plot on top of each other

```{r}
Edf_1$Empirical <- df_1$Upper_CI*100

colnames(Edf_1) <- c("E","n","Simulated","Empirical")

combo_df_1 <- melt(Edf_1, id=c("E", "n"))
colnames(combo_df_1) <- c("E","n","Data","Error")

combo_plot_1 <- ggplot(combo_df_1, aes(x=n, y=Error, colour=Data)) +
  geom_point(size=3, alpha=4/5) +
  geom_line(alpha=2/3) +
  theme_light() +
  ylim(0,50) +
  ylab("Margin of Error (Â±%)") +
  xlab("CG calls assayed") #+
  scale_colour_manual(values = get_pal("Takahe")[c(2,1)], labels = c("Empirical", "Simulated"))
combo_plot_1

#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingPlot_overlaidNEW_1.png",combo_plot_1, width=10, height = 4, dpi=1000)
```


# Low 2

# Create list for data to be dumped into
```{r}
sample_data_list_2 <- c()
```

# Sampling loop - creates random samples of sizes between 1 and n
```{r, message=F}
# Create an empty data frame to store the results
results_df_2 <- data.frame(Sample_Size = numeric(), Methylation = numeric())

for (i in c(10, 50, 100, 150, 200, 250, 300, 350, 400, 450, 500, 550, 600, 650, 700,
            750, 800, 850, 900, 950, 1000, 1050, 1100)) {
  for (j in 1:1000) {
    sample_data_2 <- sample_n(data_low_2, i)  
    methylation_table <- table(sample_data_2$MethylationCall)
    methylation <- as.numeric(methylation_table["Z"] / sum(methylation_table))
    sample_data_list_2 <- cbind(Sample_Size = i, Methylation = methylation)
    results_df_2 <- rbind(results_df_2, sample_data_list_2)
  }
}
```


# Plot
```{r}
methylation_plot_2 <- ggplot(results_df_2, aes(x=Sample_Size, y=Methylation)) +
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Methylation (%)") +
  xlab("CG calls assayed")
methylation_plot_2
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Methylation_2.png",methylation_plot_2, width=10, height = 5, dpi=1000)
```


# Finding most representative methylation call of this dataset
```{r}
mean_2 <- mean(results_df_2$Methylation[22001:23000])
```

# Making a list of the error to plot (percentage of 100%)

```{r}
results_df_2$error <- as.numeric(abs(mean_2-results_df_2$Methylation))

error_plot_2 <- ggplot(results_df_2, aes(x=Sample_Size, y=error)) + 
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Margin of Error") +
  xlab("CG calls assayed")
error_plot_2
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Error_2.png",error_plot_2, width=10, height = 5, dpi=1000)
```

# Find 99.5% confidence interval
```{r}
cis_2 <- aggregate(error ~ Sample_Size, data = results_df_2, function(x) quantile(x, c(0.0025, 0.9975)))
upper_cis_2 <- cis_2$error[,2]

df_2 <- data.frame(Sample_Size = cis_2$Sample_Size,
                 Upper_CI = upper_cis_2)
```

# Making a list of the error to plot (+/- pp)
```{r}
#establishing modelling parameters
zstar = qnorm(.995) #This defines 99.5% confidence level
p = mean_2 #The approximate methylation level from this example
n = c(10, 50, 100, 150, 200, 250, 300, 350, 400, 450, 500, 550, 600, 650, 700,
            750, 800, 850, 900, 950, 1000, 1050, 1100) #Numbers of cytosine calls assayed
E = zstar*sqrt((p*(1-p))/n) #Margin of error calculation given above variables

#coercing into dataframe
Edf_2 <- as.data.frame(E)
ndf_2 <- as.data.frame(n)
Edf_2$n <- ndf_2$n
Edf_2$Percent <- Edf_2$E*100

#establishing intercepts
line5 <- 5
line25 <- 2.5
line1 <- 1

model_plot_2 <- ggplot(Edf_2, aes(x=n, y=Percent)) +
  geom_point(alpha=2/5) +
  geom_hline(yintercept = line5, colour="#CC0000", linetype=2) +
  geom_text(aes(737,line5,label = "5%", vjust = -0.2, colour="#006600")) +
  geom_hline(yintercept = line25, colour="#006600", linetype=2) +
  geom_text(aes(737,line25,label = "2.5%", vjust = -0.2, colour="#0066CC")) +
  geom_hline(yintercept = line1, colour="#0066CC",  linetype=2) +
  geom_text(aes(737,line1,label = "1%", vjust = -0.2, colour="#CC0000")) +
  labs(x="Number of Cs in CG context", y="Margin of error") +
  theme_light() +
  ylim(0,100) +
  theme(legend.position="none")  
model_plot_2
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingModel_2.png",model_plot_2, width=10, height = 5, dpi=1000)

```


# plot on top of each other

```{r}
Edf_2$Empirical <- df_2$Upper_CI*100

colnames(Edf_2) <- c("E","n","Simulated","Empirical")

combo_df_2 <- melt(Edf_2, id=c("E", "n"))
colnames(combo_df_2) <- c("E","n","Data","Error")

combo_plot_2 <- ggplot(combo_df_2, aes(x=n, y=Error, colour=Data)) +
  geom_point(size=3, alpha=4/5) +
  geom_line(alpha=2/3) +
  theme_light() +
  ylim(0,50) +
  ylab("Margin of Error (Â±%)") +
  xlab("CG calls assayed") +
  ggtitle("47.64% Methylated") +
  scale_colour_manual(values = get_pal("Takahe")[c(2,1)], labels = c("Empirical", "Simulated"))
combo_plot_2

#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingPlot_overlaidNEW_2.png",combo_plot_2, width=10, height = 4, dpi=1000)
```

# Low 3

# Create list for data to be dumped into
```{r}
sample_data_list_3 <- c()
```

# Sampling loop - creates random samples of sizes between 1 and n
```{r, message=F}
# Create an empty data frame to store the results
results_df_3 <- data.frame(Sample_Size = numeric(), Methylation = numeric())

for (i in c(10, 50, 100, 200, 500, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000)) {
  for (j in 1:1000) {
    sample_data_3 <- sample_n(data_low_3, i)  
    methylation_table <- table(sample_data_3$MethylationCall)
    methylation <- as.numeric(methylation_table["Z"] / sum(methylation_table))
    sample_data_list_3 <- cbind(Sample_Size = i, Methylation = methylation)
    results_df_3 <- rbind(results_df_3, sample_data_list_3)
  }
}
```


# Plot
```{r}
methylation_plot_3 <- ggplot(results_df_3, aes(x=Sample_Size, y=Methylation)) +
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Methylation (%)") +
  xlab("CG calls assayed")
methylation_plot_3
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Methylation_3.png",methylation_plot_3, width=10, height = 5, dpi=1000)
```


# Finding most representative methylation call of this dataset
```{r}
mean_3 <- mean(results_df_3$Methylation[13001:14000])
```

# Making a list of the error to plot (percentage of 100%)

```{r}
results_df_3$error <- as.numeric(abs(mean_3-results_df_3$Methylation))

error_plot_3 <- ggplot(results_df_3, aes(x=Sample_Size, y=error)) + 
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Margin of Error") +
  xlab("CG calls assayed")
error_plot_3
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Error_3.png",error_plot_3, width=10, height = 5, dpi=1000)
```

# Find 99.5% confidence interval
```{r}
cis_3 <- aggregate(error ~ Sample_Size, data = results_df_3, function(x) quantile(x, c(0.0025, 0.9975)))
upper_cis_3 <- cis_3$error[,2]

df_3 <- data.frame(Sample_Size = cis_3$Sample_Size,
                 Upper_CI = upper_cis_3)
```


# Making a list of the error to plot (+/- pp)
```{r}
#establishing modelling parameters
zstar = qnorm(.995) #This defines 99.5% confidence level
p = mean_3 #The approximate methylation level from this example
n = c(10, 50, 100, 200, 500, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000) #Numbers of cytosine calls assayed
E = zstar*sqrt((p*(1-p))/n) #Margin of error calculation given above variables

#coercing into dataframe
Edf_3 <- as.data.frame(E)
ndf_3 <- as.data.frame(n)
Edf_3$n <- ndf_3$n
Edf_3$Percent <- Edf_3$E*100

#establishing intercepts
line5 <- 5
line25 <- 2.5
line1 <- 1

model_plot_3 <- ggplot(Edf_3, aes(x=n, y=Percent)) +
  geom_point(alpha=2/5) +
  geom_hline(yintercept = line5, colour="#CC0000", linetype=2) +
  geom_text(aes(737,line5,label = "5%", vjust = -0.2, colour="#006600")) +
  geom_hline(yintercept = line25, colour="#006600", linetype=2) +
  geom_text(aes(737,line25,label = "2.5%", vjust = -0.2, colour="#0066CC")) +
  geom_hline(yintercept = line1, colour="#0066CC",  linetype=2) +
  geom_text(aes(737,line1,label = "1%", vjust = -0.2, colour="#CC0000")) +
  labs(x="Number of Cs in CG context", y="Margin of error") +
  theme_light() +
  ylim(0,100) +
  theme(legend.position="none")  
model_plot_3
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingModel_3.png",model_plot_3, width=10, height = 5, dpi=1000)

```


# plot on top of each other

```{r}
Edf_3$Empirical <- df_3$Upper_CI*100

colnames(Edf_3) <- c("E","n","Simulated","Empirical")

combo_df_3 <- melt(Edf_3, id=c("E", "n"))
colnames(combo_df_3) <- c("E","n","Data","Error")

combo_plot_3 <- ggplot(combo_df_3, aes(x=n, y=Error, colour=Data)) +
  geom_point(size=3, alpha=4/5) +
  geom_line(alpha=2/3) +
  theme_light() +
  ylim(0,50) +
  ylab("Margin of Error (Â±%)") +
  xlab("CG calls assayed") +
  ggtitle("50.78% Methylated") +
  scale_colour_manual(values = get_pal("Takahe")[c(2,1)], labels = c("Empirical", "Simulated"))
combo_plot_3

#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingPlot_overlaidNEW_3.png",combo_plot_3, width=10, height = 4, dpi=1000)
```

# Low 4

# Create list for data to be dumped into
```{r}
sample_data_list_4 <- c()
```

# Sampling loop - creates random samples of sizes between 1 and n
```{r, message=F}
# Create an empty data frame to store the results
results_df_4 <- data.frame(Sample_Size = numeric(), Methylation = numeric())

for (i in c(10, 50, 100, 200, 500, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000,
            11000, 12000, 13000, 14000, 15000, 16000, 17000, 18000, 19000, 20000, 21000)) {
  for (j in 1:1000) {
    sample_data_4 <- sample_n(data_low_4, i)  
    methylation_table <- table(sample_data_4$MethylationCall)
    methylation <- as.numeric(methylation_table["Z"] / sum(methylation_table))
    sample_data_list_4 <- cbind(Sample_Size = i, Methylation = methylation)
    results_df_4 <- rbind(results_df_4, sample_data_list_4)
  }
}
```


# Plot
```{r}
methylation_plot_4 <- ggplot(results_df_4, aes(x=Sample_Size, y=Methylation)) +
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Methylation (%)") +
  xlab("CG calls assayed")
methylation_plot_4
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Methylation_4.png",methylation_plot_4, width=10, height = 5, dpi=1000)
```


# Finding most representative methylation call of this dataset
```{r}
mean_4 <- mean(results_df_4$Methylation[25001:26000])
```

# Making a list of the error to plot (percentage of 100%)

```{r}
results_df_4$error <- as.numeric(abs(mean_4-results_df_4$Methylation))

error_plot_4 <- ggplot(results_df_4, aes(x=Sample_Size, y=error)) + 
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Margin of Error") +
  xlab("CG calls assayed")
error_plot_4
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Error_4.png",error_plot_4, width=10, height = 5, dpi=1000)
```

# Find 99.5% confidence interval
```{r}
cis_4 <- aggregate(error ~ Sample_Size, data = results_df_4, function(x) quantile(x, c(0.0025, 0.9975)))
upper_cis_4 <- cis_4$error[,2]

df_4 <- data.frame(Sample_Size = cis_4$Sample_Size,
                 Upper_CI = upper_cis_4)
```


# Making a list of the error to plot (+/- pp)
```{r}
#establishing modelling parameters
zstar = qnorm(.995) #This defines 99.5% confidence level
p = mean_4 #The approximate methylation level from this example
n = c(10, 50, 100, 200, 500, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000,
            11000, 12000, 13000, 14000, 15000, 16000, 17000, 18000, 19000, 20000, 21000) #Numbers of cytosine calls assayed
E = zstar*sqrt((p*(1-p))/n) #Margin of error calculation given above variables

#coercing into dataframe
Edf_4 <- as.data.frame(E)
ndf_4 <- as.data.frame(n)
Edf_4$n <- ndf_4$n
Edf_4$Percent <- Edf_4$E*100

#establishing intercepts
line5 <- 5
line25 <- 2.5
line1 <- 1

model_plot_4 <- ggplot(Edf_4, aes(x=n, y=Percent)) +
  geom_point(alpha=2/5) +
  geom_hline(yintercept = line5, colour="#CC0000", linetype=2) +
  geom_text(aes(737,line5,label = "5%", vjust = -0.2, colour="#006600")) +
  geom_hline(yintercept = line25, colour="#006600", linetype=2) +
  geom_text(aes(737,line25,label = "2.5%", vjust = -0.2, colour="#0066CC")) +
  geom_hline(yintercept = line1, colour="#0066CC",  linetype=2) +
  geom_text(aes(737,line1,label = "1%", vjust = -0.2, colour="#CC0000")) +
  labs(x="Number of Cs in CG context", y="Margin of error") +
  theme_light() +
  ylim(0,100) +
  theme(legend.position="none")  
model_plot_4
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingModel_4.png",model_plot_4, width=10, height = 5, dpi=1000)

```


# plot on top of each other

```{r}
Edf_4$Empirical <- df_4$Upper_CI*100

colnames(Edf_4) <- c("E","n","Simulated","Empirical")

combo_df_4 <- melt(Edf_4, id=c("E", "n"))
colnames(combo_df_4) <- c("E","n","Data","Error")

combo_plot_4 <- ggplot(combo_df_4, aes(x=n, y=Error, colour=Data)) +
  geom_point(size=3, alpha=4/5) +
  geom_line(alpha=2/3) +
  theme_light() +
  ylim(0,50) +
  ylab("Margin of Error (Â±%)") +
  xlab("CG calls assayed") +
  ggtitle("52.75% Methylated") +
  scale_colour_manual(values = get_pal("Takahe")[c(2,1)], labels = c("Empirical", "Simulated"))
combo_plot_4

#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingPlot_overlaidNEW_4.png",combo_plot_4, width=10, height = 4, dpi=1000)
```

# Mid 5

# Create list for data to be dumped into
```{r}
sample_data_list_5 <- c()
```

# Sampling loop - creates random samples of sizes between 1 and n
```{r, message=F}
# Create an empty data frame to store the results
results_df_5 <- data.frame(Sample_Size = numeric(), Methylation = numeric())

for (i in c(10, 50, 100, 200, 500, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000)) {
  for (j in 1:1000) {
    sample_data_5 <- sample_n(data_mid_5, i)  
    methylation_table <- table(sample_data_5$MethylationCall)
    methylation <- as.numeric(methylation_table["Z"] / sum(methylation_table))
    sample_data_list_5 <- cbind(Sample_Size = i, Methylation = methylation)
    results_df_5 <- rbind(results_df_5, sample_data_list_5)
  }
}
```


# Plot
```{r}
methylation_plot_5 <- ggplot(results_df_5, aes(x=Sample_Size, y=Methylation)) +
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Methylation (%)") +
  xlab("CG calls assayed")
methylation_plot_5
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Methylation_5.png",methylation_plot_5, width=10, height = 5, dpi=1000)
```


# Finding most representative methylation call of this dataset
```{r}
mean_5 <- mean(results_df_5$Methylation[13001:14000])
```

# Making a list of the error to plot (percentage of 100%)

```{r}
results_df_5$error <- as.numeric(abs(mean_5-results_df_5$Methylation))

error_plot_5 <- ggplot(results_df_5, aes(x=Sample_Size, y=error)) + 
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Margin of Error") +
  xlab("CG calls assayed")
error_plot_5
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Error_5.png",error_plot_5, width=10, height = 5, dpi=1000)
```

# Find 99.5% confidence interval
```{r}
cis_5 <- aggregate(error ~ Sample_Size, data = results_df_5, function(x) quantile(x, c(0.0025, 0.9975)))
upper_cis_5 <- cis_5$error[,2]

df_5 <- data.frame(Sample_Size = cis_5$Sample_Size,
                 Upper_CI = upper_cis_5)
```


# Making a list of the error to plot (+/- pp)
```{r}
#establishing modelling parameters
zstar = qnorm(.995) #This defines 99.5% confidence level
p = mean_5 #The approximate methylation level from this example
n = c(10, 50, 100, 200, 500, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000) #Numbers of cytosine calls assayed
E = zstar*sqrt((p*(1-p))/n) #Margin of error calculation given above variables

#coercing into dataframe
Edf_5 <- as.data.frame(E)
ndf_5 <- as.data.frame(n)
Edf_5$n <- ndf_5$n
Edf_5$Percent <- Edf_5$E*100

#establishing intercepts
line5 <- 5
line25 <- 2.5
line1 <- 1

model_plot_5 <- ggplot(Edf_5, aes(x=n, y=Percent)) +
  geom_point(alpha=2/5) +
  geom_hline(yintercept = line5, colour="#CC0000", linetype=2) +
  geom_text(aes(737,line5,label = "5%", vjust = -0.2, colour="#006600")) +
  geom_hline(yintercept = line25, colour="#006600", linetype=2) +
  geom_text(aes(737,line25,label = "2.5%", vjust = -0.2, colour="#0066CC")) +
  geom_hline(yintercept = line1, colour="#0066CC",  linetype=2) +
  geom_text(aes(737,line1,label = "1%", vjust = -0.2, colour="#CC0000")) +
  labs(x="Number of Cs in CG context", y="Margin of error") +
  theme_light() +
  ylim(0,100) +
  theme(legend.position="none")  
model_plot_5
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingModel_5.png",model_plot_5, width=10, height = 5, dpi=1000)

```


# plot on top of each other

```{r}
Edf_5$Empirical <- df_5$Upper_CI*100

colnames(Edf_5) <- c("E","n","Simulated","Empirical")

combo_df_5 <- melt(Edf_5, id=c("E", "n"))
colnames(combo_df_5) <- c("E","n","Data","Error")

combo_plot_5 <- ggplot(combo_df_5, aes(x=n, y=Error, colour=Data)) +
  geom_point(size=3, alpha=4/5) +
  geom_line(alpha=2/3) +
  theme_light() +
  ylim(0,50) +
  ylab("Margin of Error (Â±%)") +
  xlab("CG calls assayed") +
  ggtitle("62.54% Methylated") +
  scale_colour_manual(values = get_pal("Takahe")[c(2,1)], labels = c("Empirical", "Simulated"))
combo_plot_5

#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingPlot_overlaidNEW_5.png",combo_plot_5, width=10, height = 4, dpi=1000)
```

# Mid 6

# Create list for data to be dumped into
```{r}
sample_data_list_6 <- c()
```

# Sampling loop - creates random samples of sizes between 1 and n
```{r, message=F}
# Create an empty data frame to store the results
results_df_6 <- data.frame(Sample_Size = numeric(), Methylation = numeric())

for (i in c(10, 50, 100, 200, 500, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000)) {
  for (j in 1:1000) {
    sample_data_6 <- sample_n(data_mid_6, i)  
    methylation_table <- table(sample_data_6$MethylationCall)
    methylation <- as.numeric(methylation_table["Z"] / sum(methylation_table))
    sample_data_list_6 <- cbind(Sample_Size = i, Methylation = methylation)
    results_df_6 <- rbind(results_df_6, sample_data_list_6)
  }
}
```


# Plot
```{r}
methylation_plot_6 <- ggplot(results_df_6, aes(x=Sample_Size, y=Methylation)) +
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Methylation (%)") +
  xlab("CG calls assayed")
methylation_plot_6
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Methylation_6.png",methylation_plot_6, width=10, height = 5, dpi=1000)
```


# Finding most representative methylation call of this dataset
```{r}
mean_6 <- mean(results_df_6$Methylation[12001:13000])
```

# Making a list of the error to plot (percentage of 100%)

```{r}
results_df_6$error <- as.numeric(abs(mean_6-results_df_6$Methylation))

error_plot_6 <- ggplot(results_df_6, aes(x=Sample_Size, y=error)) + 
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Margin of Error") +
  xlab("CG calls assayed")
error_plot_6
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Error_6.png",error_plot_6, width=10, height = 5, dpi=1000)
```

# Find 99.5% confidence interval
```{r}
cis_6 <- aggregate(error ~ Sample_Size, data = results_df_6, function(x) quantile(x, c(0.0025, 0.9975)))
upper_cis_6 <- cis_6$error[,2]

df_6 <- data.frame(Sample_Size = cis_6$Sample_Size,
                 Upper_CI = upper_cis_6)
```


# Making a list of the error to plot (+/- pp)
```{r}
#establishing modelling parameters
zstar = qnorm(.995) #This defines 99.5% confidence level
p = mean_6 #The approximate methylation level from this example
n = c(10, 50, 100, 200, 500, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000) #Numbers of cytosine calls assayed
E = zstar*sqrt((p*(1-p))/n) #Margin of error calculation given above variables

#coercing into dataframe
Edf_6 <- as.data.frame(E)
ndf_6 <- as.data.frame(n)
Edf_6$n <- ndf_6$n
Edf_6$Percent <- Edf_6$E*100

#establishing intercepts
line5 <- 5
line25 <- 2.5
line1 <- 1

model_plot_6 <- ggplot(Edf_6, aes(x=n, y=Percent)) +
  geom_point(alpha=2/5) +
  geom_hline(yintercept = line5, colour="#CC0000", linetype=2) +
  geom_text(aes(737,line5,label = "5%", vjust = -0.2, colour="#006600")) +
  geom_hline(yintercept = line25, colour="#006600", linetype=2) +
  geom_text(aes(737,line25,label = "2.5%", vjust = -0.2, colour="#0066CC")) +
  geom_hline(yintercept = line1, colour="#0066CC",  linetype=2) +
  geom_text(aes(737,line1,label = "1%", vjust = -0.2, colour="#CC0000")) +
  labs(x="Number of Cs in CG context", y="Margin of error") +
  theme_light() +
  ylim(0,100) +
  theme(legend.position="none")  
model_plot_6
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingModel_6.png",model_plot_6, width=10, height = 5, dpi=1000)

```


# plot on top of each other

```{r}
Edf_6$Empirical <- df_6$Upper_CI*100

colnames(Edf_6) <- c("E","n","Simulated","Empirical")

combo_df_6 <- melt(Edf_6, id=c("E", "n"))
colnames(combo_df_6) <- c("E","n","Data","Error")

combo_plot_6 <- ggplot(combo_df_6, aes(x=n, y=Error, colour=Data)) +
  geom_point(size=3, alpha=4/5) +
  geom_line(alpha=2/3) +
  theme_light() +
  ylim(0,50) +
  ylab("Margin of Error (Â±%)") +
  xlab("CG calls assayed") +
  ggtitle("62.90% Methylated") +
  scale_colour_manual(values = get_pal("Takahe")[c(2,1)], labels = c("Empirical", "Simulated"))
combo_plot_6

#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingPlot_overlaidNEW_6.png",combo_plot_6, width=10, height = 4, dpi=1000)
```

# Mid 7

# Create list for data to be dumped into
```{r}
sample_data_list_7 <- c()
```

# Sampling loop - creates random samples of sizes between 1 and n
```{r, message=F}
# Create an empty data frame to store the results
results_df_7 <- data.frame(Sample_Size = numeric(), Methylation = numeric())

for (i in c(10, 50, 100, 200, 500, 1000, 1500, 2000, 2500, 3000, 3500, 4000, 4500)) {
  for (j in 1:1000) {
    sample_data_7 <- sample_n(data_mid_7, i)  
    methylation_table <- table(sample_data_7$MethylationCall)
    methylation <- as.numeric(methylation_table["Z"] / sum(methylation_table))
    sample_data_list_7 <- cbind(Sample_Size = i, Methylation = methylation)
    results_df_7 <- rbind(results_df_7, sample_data_list_7)
  }
}
```


# Plot
```{r}
methylation_plot_7 <- ggplot(results_df_7, aes(x=Sample_Size, y=Methylation)) +
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Methylation (%)") +
  xlab("CG calls assayed")
methylation_plot_7
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Methylation_7.png",methylation_plot_7, width=10, height = 5, dpi=1000)
```


# Finding most representative methylation call of this dataset
```{r}
mean_7 <- mean(results_df_7$Methylation[12001:13000])
```

# Making a list of the error to plot (percentage of 100%)

```{r}
results_df_7$error <- as.numeric(abs(mean_7-results_df_7$Methylation))

error_plot_7 <- ggplot(results_df_7, aes(x=Sample_Size, y=error)) + 
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Margin of Error") +
  xlab("CG calls assayed")
error_plot_7
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Error_7.png",error_plot_7, width=10, height = 5, dpi=1000)
```

# Find 99.5% confidence interval
```{r}
cis_7 <- aggregate(error ~ Sample_Size, data = results_df_7, function(x) quantile(x, c(0.0025, 0.9975)))
upper_cis_7 <- cis_7$error[,2]

df_7 <- data.frame(Sample_Size = cis_7$Sample_Size,
                 Upper_CI = upper_cis_7)
```


# Making a list of the error to plot (+/- pp)
```{r}
#establishing modelling parameters
zstar = qnorm(.995) #This defines 99.5% confidence level
p = mean_7 #The approximate methylation level from this example
n = c(10, 50, 100, 200, 500, 1000, 1500, 2000, 2500, 3000, 3500, 4000, 4500) #Numbers of cytosine calls assayed
E = zstar*sqrt((p*(1-p))/n) #Margin of error calculation given above variables

#coercing into dataframe
Edf_7 <- as.data.frame(E)
ndf_7 <- as.data.frame(n)
Edf_7$n <- ndf_7$n
Edf_7$Percent <- Edf_7$E*100

#establishing intercepts
line5 <- 5
line25 <- 2.5
line1 <- 1

model_plot_7 <- ggplot(Edf_7, aes(x=n, y=Percent)) +
  geom_point(alpha=2/5) +
  geom_hline(yintercept = line5, colour="#CC0000", linetype=2) +
  geom_text(aes(737,line5,label = "5%", vjust = -0.2, colour="#006600")) +
  geom_hline(yintercept = line25, colour="#006600", linetype=2) +
  geom_text(aes(737,line25,label = "2.5%", vjust = -0.2, colour="#0066CC")) +
  geom_hline(yintercept = line1, colour="#0066CC",  linetype=2) +
  geom_text(aes(737,line1,label = "1%", vjust = -0.2, colour="#CC0000")) +
  labs(x="Number of Cs in CG context", y="Margin of error") +
  theme_light() +
  ylim(0,100) +
  theme(legend.position="none")  
model_plot_7
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingModel_7.png",model_plot_7, width=10, height = 5, dpi=1000)

```


# plot on top of each other

```{r}
Edf_7$Empirical <- df_7$Upper_CI*100

colnames(Edf_7) <- c("E","n","Simulated","Empirical")

combo_df_7 <- melt(Edf_7, id=c("E", "n"))
colnames(combo_df_7) <- c("E","n","Data","Error")

combo_plot_7 <- ggplot(combo_df_7, aes(x=n, y=Error, colour=Data)) +
  geom_point(size=3, alpha=4/5) +
  geom_line(alpha=2/3) +
  theme_light() +
  ylim(0,50) +
  ylab("Margin of Error (Â±%)") +
  xlab("CG calls assayed") +
  ggtitle("63.21% Methylated") +
  scale_colour_manual(values = get_pal("Takahe")[c(2,1)], labels = c("Empirical", "Simulated"))
combo_plot_7

#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingPlot_overlaidNEW_7.png",combo_plot_7, width=10, height = 4, dpi=1000)
```

# High 8

# Create list for data to be dumped into
```{r}
sample_data_list_8 <- c()
```

# Sampling loop - creates random samples of sizes between 1 and n
```{r, message=F}
# Create an empty data frame to store the results
results_df_8 <- data.frame(Sample_Size = numeric(), Methylation = numeric())

for (i in c(10, 50, 100, 200, 500, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000)) {
  for (j in 1:1000) {
    sample_data_8 <- sample_n(data_high_8, i)  
    methylation_table <- table(sample_data_8$MethylationCall)
    methylation <- as.numeric(methylation_table["Z"] / sum(methylation_table))
    sample_data_list_8 <- cbind(Sample_Size = i, Methylation = methylation)
    results_df_8 <- rbind(results_df_8, sample_data_list_8)
  }
}
```


# Plot
```{r}
methylation_plot_8 <- ggplot(results_df_8, aes(x=Sample_Size, y=Methylation)) +
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Methylation (%)") +
  xlab("CG calls assayed")
methylation_plot_8
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Methylation_8.png",methylation_plot_8, width=10, height = 5, dpi=1000)
```


# Finding most representative methylation call of this dataset
```{r}
mean_8 <- mean(results_df_8$Methylation[13001:14000])
```

# Making a list of the error to plot (percentage of 100%)

```{r}
results_df_8$error <- as.numeric(abs(mean_8-results_df_8$Methylation))

error_plot_8 <- ggplot(results_df_8, aes(x=Sample_Size, y=error)) + 
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Margin of Error") +
  xlab("CG calls assayed")
error_plot_8
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Error_8.png",error_plot_8, width=10, height = 5, dpi=1000)
```

# Find 99.5% confidence interval
```{r}
cis_8 <- aggregate(error ~ Sample_Size, data = results_df_8, function(x) quantile(x, c(0.0025, 0.9975)))
upper_cis_8 <- cis_8$error[,2]

df_8 <- data.frame(Sample_Size = cis_8$Sample_Size,
                 Upper_CI = upper_cis_8)
```


# Making a list of the error to plot (+/- pp)
```{r}
#establishing modelling parameters
zstar = qnorm(.995) #This defines 99.5% confidence level
p = mean_8 #The approximate methylation level from this example
n = c(10, 50, 100, 200, 500, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000) #Numbers of cytosine calls assayed
E = zstar*sqrt((p*(1-p))/n) #Margin of error calculation given above variables

#coercing into dataframe
Edf_8 <- as.data.frame(E)
ndf_8 <- as.data.frame(n)
Edf_8$n <- ndf_8$n
Edf_8$Percent <- Edf_8$E*100

#establishing intercepts
line5 <- 5
line25 <- 2.5
line1 <- 1

model_plot_8 <- ggplot(Edf_8, aes(x=n, y=Percent)) +
  geom_point(alpha=2/5) +
  geom_hline(yintercept = line5, colour="#CC0000", linetype=2) +
  geom_text(aes(737,line5,label = "5%", vjust = -0.2, colour="#006600")) +
  geom_hline(yintercept = line25, colour="#006600", linetype=2) +
  geom_text(aes(737,line25,label = "2.5%", vjust = -0.2, colour="#0066CC")) +
  geom_hline(yintercept = line1, colour="#0066CC",  linetype=2) +
  geom_text(aes(737,line1,label = "1%", vjust = -0.2, colour="#CC0000")) +
  labs(x="Number of Cs in CG context", y="Margin of error") +
  theme_light() +
  ylim(0,100) +
  theme(legend.position="none")  
model_plot_8
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingModel_8.png",model_plot_8, width=10, height = 5, dpi=1000)

```


# plot on top of each other

```{r}
Edf_8$Empirical <- df_8$Upper_CI*100

colnames(Edf_8) <- c("E","n","Simulated","Empirical")

combo_df_8 <- melt(Edf_8, id=c("E", "n"))
colnames(combo_df_8) <- c("E","n","Data","Error")

combo_plot_8 <- ggplot(combo_df_8, aes(x=n, y=Error, colour=Data)) +
  geom_point(size=3, alpha=4/5) +
  geom_line(alpha=2/3) +
  theme_light() +
  ylim(0,50) +
  ylab("Margin of Error (Â±%)") +
  xlab("CG calls assayed") +
  ggtitle("82.90% Methylated") +
  scale_colour_manual(values = get_pal("Takahe")[c(2,1)], labels = c("Empirical", "Simulated"))
combo_plot_8

#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingPlot_overlaidNEW_8.png",combo_plot_8, width=10, height = 4, dpi=1000)
```

# High 9

# Create list for data to be dumped into
```{r}
sample_data_list_9 <- c()
```

# Sampling loop - creates random samples of sizes between 1 and n
```{r, message=F}
# Create an empty data frame to store the results
results_df_9 <- data.frame(Sample_Size = numeric(), Methylation = numeric())

for (i in c(10, 50, 100, 200, 500, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000)) {
  for (j in 1:1000) {
    sample_data_9 <- sample_n(data_high_9, i)  
    methylation_table <- table(sample_data_9$MethylationCall)
    methylation <- as.numeric(methylation_table["Z"] / sum(methylation_table))
    sample_data_list_9 <- cbind(Sample_Size = i, Methylation = methylation)
    results_df_9 <- rbind(results_df_9, sample_data_list_9)
  }
}
```


# Plot
```{r}
methylation_plot_9 <- ggplot(results_df_9, aes(x=Sample_Size, y=Methylation)) +
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Methylation (%)") +
  xlab("CG calls assayed")
methylation_plot_9
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Methylation_9.png",methylation_plot_9, width=10, height = 5, dpi=1000)
```


# Finding most representative methylation call of this dataset
```{r}
mean_9 <- mean(results_df_9$Methylation[12001:13000])
```

# Making a list of the error to plot (percentage of 100%)

```{r}
results_df_9$error <- as.numeric(abs(mean_9-results_df_9$Methylation))

error_plot_9 <- ggplot(results_df_9, aes(x=Sample_Size, y=error)) + 
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Margin of Error") +
  xlab("CG calls assayed")
error_plot_9
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Error_9.png",error_plot_9, width=10, height = 5, dpi=1000)
```

# Find 99.5% confidence interval
```{r}
cis_9 <- aggregate(error ~ Sample_Size, data = results_df_9, function(x) quantile(x, c(0.0025, 0.9975)))
upper_cis_9 <- cis_9$error[,2]

df_9 <- data.frame(Sample_Size = cis_9$Sample_Size,
                 Upper_CI = upper_cis_9)
```


# Making a list of the error to plot (+/- pp)
```{r}
#establishing modelling parameters
zstar = qnorm(.995) #This defines 99.5% confidence level
p = mean_9 #The approximate methylation level from this example
n = c(10, 50, 100, 200, 500, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000) #Numbers of cytosine calls assayed
E = zstar*sqrt((p*(1-p))/n) #Margin of error calculation given above variables

#coercing into dataframe
Edf_9 <- as.data.frame(E)
ndf_9 <- as.data.frame(n)
Edf_9$n <- ndf_9$n
Edf_9$Percent <- Edf_9$E*100

#establishing intercepts
line5 <- 5
line25 <- 2.5
line1 <- 1

model_plot_9 <- ggplot(Edf_9, aes(x=n, y=Percent)) +
  geom_point(alpha=2/5) +
  geom_hline(yintercept = line5, colour="#CC0000", linetype=2) +
  geom_text(aes(737,line5,label = "5%", vjust = -0.2, colour="#006600")) +
  geom_hline(yintercept = line25, colour="#006600", linetype=2) +
  geom_text(aes(737,line25,label = "2.5%", vjust = -0.2, colour="#0066CC")) +
  geom_hline(yintercept = line1, colour="#0066CC",  linetype=2) +
  geom_text(aes(737,line1,label = "1%", vjust = -0.2, colour="#CC0000")) +
  labs(x="Number of Cs in CG context", y="Margin of error") +
  theme_light() +
  ylim(0,100) +
  theme(legend.position="none")  
model_plot_9
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingModel_9.png",model_plot_9, width=10, height = 5, dpi=1000)

```


# plot on top of each other

```{r}
Edf_9$Empirical <- df_9$Upper_CI*100

colnames(Edf_9) <- c("E","n","Simulated","Empirical")

combo_df_9 <- melt(Edf_9, id=c("E", "n"))
colnames(combo_df_9) <- c("E","n","Data","Error")

combo_plot_9 <- ggplot(combo_df_9, aes(x=n, y=Error, colour=Data)) +
  geom_point(size=3, alpha=4/5) +
  geom_line(alpha=2/3) +
  theme_light() +
  ylim(0,50) +
  ylab("Margin of Error (Â±%)") +
  xlab("CG calls assayed") +
  ggtitle("80.88% Methylated") +
  scale_colour_manual(values = get_pal("Takahe")[c(2,1)], labels = c("Empirical", "Simulated"))
combo_plot_9

#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingPlot_overlaidNEW_9.png",combo_plot_9, width=10, height = 4, dpi=1000)
```

# High 10

# Create list for data to be dumped into
```{r}
sample_data_list_10 <- c()
```

# Sampling loop - creates random samples of sizes between 1 and n
```{r, message=F}
# Create an empty data frame to store the results
results_df_10 <- data.frame(Sample_Size = numeric(), Methylation = numeric())

for (i in c(10, 50, 100, 200, 500, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000, 11000, 12000, 13000, 14000, 15000, 16000)) {
  for (j in 1:1000) {
    sample_data_10 <- sample_n(data_high_10, i)  
    methylation_table <- table(sample_data_10$MethylationCall)
    methylation <- as.numeric(methylation_table["Z"] / sum(methylation_table))
    sample_data_list_10 <- cbind(Sample_Size = i, Methylation = methylation)
    results_df_10 <- rbind(results_df_10, sample_data_list_10)
  }
}
```


# Plot
```{r}
methylation_plot_10 <- ggplot(results_df_10, aes(x=Sample_Size, y=Methylation)) +
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Methylation (%)") +
  xlab("CG calls assayed")
methylation_plot_10
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Methylation_10.png",methylation_plot_10, width=10, height = 5, dpi=1000)
```


# Finding most representative methylation call of this dataset
```{r}
mean_10 <- mean(results_df_10$Methylation[20001:21000])
```

# Making a list of the error to plot (percentage of 100%)

```{r}
results_df_10$error <- as.numeric(abs(mean_10-results_df_10$Methylation))

error_plot_10 <- ggplot(results_df_10, aes(x=Sample_Size, y=error)) + 
  geom_point(size=3,alpha=1/3) +
  theme_light() +
  ylab("Margin of Error") +
  xlab("CG calls assayed")
error_plot_10
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/EmpiricalSampling_Error_10.png",error_plot_10, width=10, height = 5, dpi=1000)
```

# Find 99.5% confidence interval
```{r}
cis_10 <- aggregate(error ~ Sample_Size, data = results_df_10, function(x) quantile(x, c(0.0025, 0.9975)))
upper_cis_10 <- cis_10$error[,2]

df_10 <- data.frame(Sample_Size = cis_10$Sample_Size,
                 Upper_CI = upper_cis_10)
```


# Making a list of the error to plot (+/- pp)
```{r}
#establishing modelling parameters
zstar = qnorm(.995) #This defines 99.5% confidence level
p = mean_10 #The approximate methylation level from this example
n = c(10, 50, 100, 200, 500, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000, 11000, 12000, 13000, 14000, 15000, 16000) #Numbers of cytosine calls assayed
E = zstar*sqrt((p*(1-p))/n) #Margin of error calculation given above variables

#coercing into dataframe
Edf_10 <- as.data.frame(E)
ndf_10 <- as.data.frame(n)
Edf_10$n <- ndf_10$n
Edf_10$Percent <- Edf_10$E*100

#establishing intercepts
line5 <- 5
line25 <- 2.5
line1 <- 1

model_plot_10 <- ggplot(Edf_10, aes(x=n, y=Percent)) +
  geom_point(alpha=2/5) +
  geom_hline(yintercept = line5, colour="#CC0000", linetype=2) +
  geom_text(aes(737,line5,label = "5%", vjust = -0.2, colour="#006600")) +
  geom_hline(yintercept = line25, colour="#006600", linetype=2) +
  geom_text(aes(737,line25,label = "2.5%", vjust = -0.2, colour="#0066CC")) +
  geom_hline(yintercept = line1, colour="#0066CC",  linetype=2) +
  geom_text(aes(737,line1,label = "1%", vjust = -0.2, colour="#CC0000")) +
  labs(x="Number of Cs in CG context", y="Margin of error") +
  theme_light() +
  ylim(0,100) +
  theme(legend.position="none")  
model_plot_10
#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingModel_10.png",model_plot_10, width=10, height = 5, dpi=1000)

```


# plot on top of each other

```{r}
Edf_10$Empirical <- df_10$Upper_CI*100

colnames(Edf_10) <- c("E","n","Simulated","Empirical")

combo_df_10 <- melt(Edf_10, id=c("E", "n"))
colnames(combo_df_10) <- c("E","n","Data","Error")

combo_plot_10 <- ggplot(combo_df_10, aes(x=n, y=Error, colour=Data)) +
  geom_point(size=3, alpha=4/5) +
  geom_line(alpha=2/3) +
  theme_light() +
  ylim(0,50) +
  ylab("Margin of Error (Â±%)") +
  xlab("CG calls assayed") +
  ggtitle("82.90% Methylated") +
  scale_colour_manual(values = get_pal("Takahe")[c(2,1)], labels = c("Empirical", "Simulated"))
combo_plot_10

#ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingPlot_overlaidNEW_10.png",combo_plot_10, width=10, height = 4, dpi=1000)
```

```{r}
#combo_plot_1_leg <- combo_plot_1 + theme(legend.position="none")
combo_plot_2_leg <- combo_plot_2 + theme(legend.position="none")
combo_plot_3_leg <- combo_plot_3 + theme(legend.position="none")
combo_plot_4_leg <- combo_plot_4 + theme(legend.position="none")
combo_plot_5_leg <- combo_plot_5 + theme(legend.position="none")
combo_plot_6_leg <- combo_plot_6 + theme(legend.position="none")
combo_plot_7_leg <- combo_plot_7 + theme(legend.position="none")
combo_plot_8_leg <- combo_plot_8 + theme(legend.position="none")
combo_plot_9_leg <- combo_plot_9 + theme(legend.position="none")
combo_plot_10_leg <- combo_plot_10 + theme(legend.position="none")

comboplot_low  <- cowplot::plot_grid(combo_plot_2_leg, combo_plot_3_leg, combo_plot_4_leg, nrow=1)
comboplot_mid <- cowplot::plot_grid(combo_plot_5_leg, combo_plot_6_leg, combo_plot_7_leg, nrow=1)
comboplot_high <- cowplot::plot_grid(combo_plot_10_leg, combo_plot_9_leg, combo_plot_8_leg, nrow=1)

legend_b <- cowplot::get_legend(
  combo_plot_2 + 
    guides(color = guide_legend(nrow = 1)) +
    theme(legend.position = "bottom")
)


combocomboplot <- cowplot::plot_grid(comboplot_low, comboplot_mid, comboplot_high, legend_b, rel_heights = c(1,1,1,0.1), nrow=4, labels=c("A", "B", "C"))
combocomboplot

ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SamplingPlot_overlaid_all10samples_NewMay.png",combocomboplot, width=7, height = 8, dpi=1000)

```



