---
title: "Sheep Androgen Clock Construction - Array"
author: "Victoria Sugrue"
date: 'Last updated: 30 May 2023'
output:
  word_document:
    toc: yes
    toc_depth: '6'
  html_document:
    toc: yes
    toc_float: yes
    toc_depth: 6
  pdf_document:
    toc: yes
    toc_depth: 5
    df_print: kable
    latex_engine: xelatex
subtitle: ' '
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Load required packages
```{r}
library(ggplot2)
library(caret)
library(dplyr)
library(Manu)
library(knitr)
library(sjPlot)
library(lmtest)
library(Amelia)
library(mlbench)
library(ggfortify)
```

# Data preparation
```{r}
#read in minfi normalised beta values
sheep_array_data2 <- read.csv("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Bioinformatics/Data/top10asDMP.csv", header=FALSE)

#transpose the dataframe and remove unneeded columns
sheep_array_data <- as.data.frame(t(sheep_array_data2[,-c(1:2)]))

#read in file for headers and transpose it, assign the headers to the minfi normalised data
headers <- read.csv("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Bioinformatics/Data/top10asDMP_headers.csv", header=FALSE)
theaders <- as.list(t(headers))
names(sheep_array_data) <- theaders

sheep_array_data_ear <- sheep_array_data %>% filter(!(Sex=="Blood"))



# Checking for missingness
#pdf("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/sheepdata_missingness.pdf") 
missmap(sheep_array_data, col=c("blue", "red"), legend=T)
missmap(sheep_array_data_ear, col=c("blue", "red"), legend=T)

#dev.off()

# Setting AndroAge with predicted time of puberty onset = age minus 0.5 years (i.e estimate to reach puberty at 6 months)
sheep_array_data_ear$YPMP <- as.numeric(sheep_array_data_ear$Age)-0.5
sheep_array_data_ear$YPMP[sheep_array_data_ear$Sex=="Wether"] <- 0
sheep_array_data_ear$YPMP[sheep_array_data_ear$Sex=="Ewe"] <- 0

# Subset by sex/castration status
ram_array_ear_data <- sheep_array_data_ear %>% filter(Sex=="Ram")
wether_array_ear_data <- sheep_array_data_ear %>% filter(Sex=="Wether")
ewe_array_ear_data <- sheep_array_data_ear %>% filter(Sex=="Ewe")

```

# A birds-eye view of the data 

```{r}
# Age distribution
age_dist_ARRAY <- ggplot(sheep_array_data_ear, aes(x = as.numeric(Age))) +
  geom_histogram(aes(fill = Sex), position = "dodge", alpha = 1, bins = 20) +
  scale_fill_manual(values = get_pal("Takahe")[c(1,2,4)]) +
  theme_light() +
  labs(title = "Distribution of age by sex (Sheep ear, Array)", fill = "Sex", x="Age (Years)") +
  ylab("Frequency")
age_dist_ARRAY

ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/age_distribution_clockSheep_ARRAY.png",age_dist_ARRAY, width=10, height = 8, dpi=1000)
```

# Plotting methylation data

```{r}
MKLN1_sheep_ARRAY <- ggplot(sheep_array_data_ear, aes(x=as.numeric(Age), y=as.numeric(cg21524116)*100, colour=Sex)) +
  geom_jitter(width=0.1, size=4, alpha=2/3) +
  geom_smooth(method="lm") +
  scale_colour_manual(values = get_pal("Takahe")[c(1,2,4)]) +
  theme_light() +
  labs(title = "Methylation at cg21524116, Array", colour="Sex", y="Methylation (%)", x="Age (Years)")
MKLN1_sheep_ARRAY

ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/MKLN1_meth_sheepear_ARRAY.png",MKLN1_sheep_ARRAY, width=10, height = 8, dpi=1000)
```

# While I'm here, plotting MKLN1 in sheep blood

```{r}
sheep_array_data_blood <- sheep_array_data %>% filter(Sex=="Blood")

# best to only plot non-transgenic
sheep_array_data_blood_WT <- sheep_array_data_blood %>% filter(GENOTYPE=="Control")

MKLN1_sheep_ARRAY_BLOOD_WT <- ggplot(sheep_array_data_blood_WT, aes(x=as.numeric(Age), y=as.numeric(cg21524116)*100, colour=BloodSex)) +
  geom_jitter(width=0.1, size=4, alpha=2/3) +
  geom_smooth(method="lm") +
  scale_colour_manual(values = get_pal("Takahe")[c(1,2,4)]) +
  theme_light() +
  labs(title = "Methylation at cg21524116, Array, Blood, WT only", colour="Sex", y="Methylation (%)", x="Age (Years)")
MKLN1_sheep_ARRAY_BLOOD_WT

ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/MKLN1_meth_sheepear_ARRAY_BLOOD_WT.png",MKLN1_sheep_ARRAY_BLOOD_WT, width=10, height = 8, dpi=1000)
```

# Fitting model to array data, sheep ear only

```{r}
# Run model, array data, training on rams
model_array1 <- lm(YPMP ~ as.numeric(cg21524116), data=ram_array_ear_data)
summary(model_array1)
```

# Diagnostic plots

```{r}
pdf("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/Model_array_1_ggplot_diagnostics_white.pdf", width=10, height=8)

autoplot(model_array1, which=c(1,2,3,5), size=3, alpha=4/5, colour="#313657", smooth.colour="#DD3C51", smooth.linetype="dashed", label.size = 3) + theme_light() + theme(plot.title = element_text(color = "white"))

dev.off()
```




```{r}
fitted_array <- as.data.frame(model_array1$fitted.values)
fitted_array["YPMP"] <- ram_array_ear_data$YPMP
fitted_array$YPMP <- as.numeric(fitted_array$YPMP)
fitted_array$Fit_AndroAge <- as.numeric(fitted_array$`model_array1$fitted.values`)
fitted_array$Age <- ram_array_ear_data$Age
```

# Calculating error
## All

```{r}
MAE_model_array1 <- as.data.frame(Metrics::mae(actual=fitted_array$YPMP, predicted=fitted_array$`model_array1$fitted.values`))
MAE_model_array1 # 0.496
```

```{r}
cor(fitted_array$YPMP, fitted_array$Fit_AndroAge, method="pearson")
```


# Testing the model fit further: 

# Test density/normality of residuals - looking for normally distributed. 
```{r}
d<-density(model_array1[['residuals']])
plot(d,main='Residual KDE Plot',xlab='Residual value')
```

# Breusch-Pagan test of variance (homo/heteroscedasticity)

```{r}
plot(model_array1$residuals)
bptest(model_array1)
```

# 0.4728
# p value >0.05 suggests that we have homoscedasticity. good! 

# Shapiro-Wilk test of normality

```{r}
shapiro.test(model_array1$residuals)
```

# 9.55e-05
# p value >0.05 suggests data is not normally distributed. not good!

# Leave-one-out cross validation

```{r}
ram_array_ear_data$cg21524116 <- as.numeric(ram_array_ear_data$cg21524116)
ram_array_ear_data$YPMP <- as.numeric(ram_array_ear_data$YPMP)
ctrl <- trainControl(method="LOOCV")
RamLOO_ARRAY <- train(YPMP ~ cg21524116, data=ram_array_ear_data, method="lm", trControl=ctrl)
summary(RamLOO_ARRAY)
```

```{r}
print(RamLOO_ARRAY)
```

```{r}
fittedLOO_ARRAY <- as.data.frame(RamLOO_ARRAY$pred)
fittedLOO_ARRAY["YPMP"] <- ram_array_ear_data$YPMP

RamLOO_ARRAY_MAE <- as.data.frame(Metrics::mae(actual=fittedLOO_ARRAY$YPMP, predicted=fittedLOO_ARRAY$pred))
RamLOO_ARRAY_MAE # 0.517
```

```{r}
cor(fittedLOO_ARRAY$obs, fittedLOO_ARRAY$pred, method="pearson")



ramLOOfit_array <- ggplot(fittedLOO_ARRAY, aes(x=obs, y=pred)) +
  geom_point(size=4, alpha=3/4, colour="#313657") + 
  geom_smooth(method="lm", se=F, colour="#DD3C51") +
  geom_abline(linetype="dashed") +
  theme_light() +
  ylab("Model AndroAge (Years)") +
  xlab("Years Post Male Puberty (YPMP)") +
  labs(title = "Array; Leave-one-out",
              subtitle = "MAE=0.517, cor=0.900") +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(plot.subtitle = element_text(hjust = 0.5))
ramLOOfit_array
ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/arrayerrormodel_LOO_ramFit.png",ramLOOfit_array, width=8, height = 7, dpi=1000)
```




# Now, to include ewes and wethers.

```{r}
ewe_array_ear_data$Model_AndroAge <- predict(model_array1, newdata = ewe_array_ear_data)

ewe_MAE_ARRAY <- as.data.frame(Metrics::mae(actual=ewe_array_ear_data$YPMP, predicted=ewe_array_ear_data$Model_AndroAge))
ewe_MAE_ARRAY
# Ewe MAE = 0.263 years
```


```{r}
wether_array_ear_data$Model_AndroAge <- predict(model_array1, newdata = wether_array_ear_data)

wether_MAE_ARRAY <- as.data.frame(Metrics::mae(actual=wether_array_ear_data$YPMP, predicted=wether_array_ear_data$Model_AndroAge))
wether_MAE_ARRAY
# Wether MAE = 0.363 years

ewewether_array <- rbind(ewe_array_ear_data, wether_array_ear_data)

ewewether_array_violin <- ggplot(ewewether_array, aes(x=Sex, y=Model_AndroAge, fill=Sex)) +
  geom_violin() + 
  geom_point(size=4, alpha=3/4, shape=1) +
  theme_light() +
    scale_fill_manual(values = get_pal("Takahe")[c(1,4)]) +
  geom_hline(yintercept = 0, linetype="dashed") +
  ylab("AndroAge predicted by model (Years)") +
  xlab("Sex") +
  ylab("Model AndroAge (Years)") +
  xlab("Sex") +
  labs(title = "Ewe and wether androgen clock fit (Array), trained on all rams",
              subtitle = "Ewe MAE=0.263, Wether MAE=0.363") +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(plot.subtitle = element_text(hjust = 0.5)) +
  theme(legend.position = "none")
ewewether_array_violin

array_MAE_cowplot <- cowplot::plot_grid(ramfit_array, ramLOOfit_array, labels=c("A", "B"), nrow=1)
array_MAE_cowplot
array_MAE_cowplot2 <- cowplot::plot_grid(array_MAE_cowplot, ewewether_array_violin, labels=c(" ", "C"), nrow=2)
array_MAE_cowplot2

ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/arraymodel_MAE_ALL.png",array_MAE_cowplot2, width=10, height = 10, dpi=1000)

```


```{r}
ram_array_ear_data$Model_AndroAge <- predict(model_array1, newdata = ram_array_ear_data)

combined_array <- rbind(ram_array_ear_data, wether_array_ear_data, ewe_array_ear_data)

combined_array$AndroAgeAccel <- combined_array$Model_AndroAge-combined_array$YPMP

write.csv(combined_array, "/Users/victoriasugrue/Dropbox (Otago University)/combined_array.csv")
d2combined_array <- read.csv("/Users/victoriasugrue/Dropbox (Otago University)/d2_combined_array.csv")

SheepAndroAgeAccel_Array <- ggplot(d2combined_array, aes(x=Group, y=mean)) + 
  geom_bar(stat = "identity", fill="lightgrey", colour="black") + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=0.5) + 
  geom_jitter(combined_array, mapping = aes(x=Sex, y=AndroAgeAccel, colour=Sex), size=4, alpha=1/2, width=0.05) +
  ylab("AndroAge Acceleration (Years)") + theme_classic() + geom_hline(yintercept=0) + 
  theme_light() +
  scale_colour_manual(values = get_pal("Takahe")[c(1,2,4)]) +
  theme(legend.position="none") 
SheepAndroAgeAccel_Array
ggsave("/Users/victoriasugrue/Dropbox (Otago University)/00_ThesisWriting/Figures/SheepAndroAgeAccel_Array.png", SheepAndroAgeAccel_Array, width=5,height=4, dpi=1000)
```
